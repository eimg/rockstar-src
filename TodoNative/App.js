import React from 'react';
import {
    StyleSheet,
    Text,
    View,
    ToolbarAndroid,
    FlatList,
    TouchableOpacity,
    TextInput,
    Button,
} from 'react-native';
import {
    MaterialIcons,
} from '@expo/vector-icons';

const styles = StyleSheet.create({
    toolbar: {
        padding: 20,
        paddingTop: 40,
        backgroundColor: 'steelblue',
        elevation: 4,
        flexDirection: 'row'
    },
    title: {
        fontSize: 20,
        color: 'white',
        marginLeft: 25,
        flex: 10
    },
    task: {
        padding: 20,
        borderBottomWidth: 1,
        borderBottomColor: '#ccc',
        flexDirection: 'row'
    },
    text: {
        fontSize: 18,
        flex: 10
    },
    splitter: {
        fontWeight: 'bold',
        margin: 20,
        color: '#666'
    },
    add: {
        margin: 20,
        flexDirection: 'row'
    },
    input: {
        borderColor: 'steelblue',
        borderWidth: 1,
        padding: 8,
        flex: 10
    }
});

class App extends React.Component {
    currentKey = 3;

    state = {
        todo: [
            { key: '1', subject: 'Milk', status: 0 },
            { key: '2', subject: 'Apple', status: 1 },
            { key: '3', subject: 'Bread', status: 0 },
        ],
        text: ''
    }

    componentDidMount () {
        fetch('http://206.189.42.134/tasks').then((res) => res.json()).then((json) => {
            this.setState({
                todo: json
            })
        });
    }

    add = () => {
        var newKey = ++this.currentKey + '';
        this.setState({
            todo: [
                ...this.state.todo,
                { key: newKey, subject: this.state.text, status: 0 }
            ],
            text: ''
        });

        fetch(`http://206.189.42.134/tasks`, {
            method: 'post',
            headers: {
                "Content-Type": "application/json",
            },
            body: JSON.stringify({
                subject: this.state.text, status: 0
            })
        });
    }

    clear = () => {
        this.setState({
            todo: this.state.todo.filter((task) => task.status === 0)
        })
    }

    remove = (key) => {
        this.setState({
            todo: this.state.todo.filter((task) => task.key !== key)
        })
    }

    done = (key) => {
        this.setState({
            todo: this.state.todo.map((task) => {
                if(task.key === key) task.status = 1;
                return task;
            })
        })
    }

    undo = (key) => {
        this.setState({
            todo: this.state.todo.map((task) => {
                if(task.key === key) task.status = 0;
                return task;
            })
        })
    }

    renderItem = ({item}) => {
        return (
            <View style={styles.task}>
                {item.status === 0 ? (
                    <TouchableOpacity style={{flex: 2}} onPress={() => {
                        this.done(item.key)
                    }}>
                        <MaterialIcons
                            name="check-box-outline-blank" size={24} />
                    </TouchableOpacity>
                ) : (
                    <TouchableOpacity style={{flex: 2}} onPress={() => {
                        this.undo(item.key)
                    }}>
                        <MaterialIcons
                            name="check-box" size={24} />
                    </TouchableOpacity>
                )}
                <Text style={styles.text}>{item.subject}</Text>
                <TouchableOpacity style={{flex: 1}} onPress={()=>{
                    this.remove(item.key)
                }}>
                    <MaterialIcons name="delete" size={24} />
                </TouchableOpacity>
            </View>
        )
    }

    keyExtractor = (item, index) => item._id

    render() {
        return (
            <View>
                <View style={styles.toolbar}>
                    <MaterialIcons name="done-all" size={32} color="white" />
                    <Text style={styles.title}>Todo Native</Text>
                    <TouchableOpacity onPress={this.clear}>
                        <MaterialIcons name="clear-all" size={32} color="white" />
                    </TouchableOpacity>
                </View>

                <View style={styles.add}>
                    <TextInput
                        style={styles.input}
                        placeholder="Enter your task"
                        onChangeText={(text) => {
                            this.setState({ text: text })
                        }}
                        value={this.state.text}
                    />
                    <Button title="Add" onPress={this.add} />
                </View>

                <FlatList
                    data={this.state.todo.filter(task=>task.status === 0)}
                    renderItem={this.renderItem}
                    keyExtractor={this.keyExtractor}
                />
                <View><Text style={styles.splitter}>Done</Text></View>
                <FlatList
                    data={this.state.todo.filter(task=>task.status === 1)}
                    renderItem={this.renderItem}
                    keyExtractor={this.keyExtractor}
                />
            </View>
        );
    }
}

export default App;
