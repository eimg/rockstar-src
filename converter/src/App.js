import React, { Component } from 'react';
import { mmToMetric, mmToImperial, metricToMm, imperialToMm, } from './converter';
import Header from './Header';

class App extends Component {
    state = { peittha: 1, kg: 1.63, lb: 3.6 };

    handleChange = (e) => {
        var value = e.target.value;
        var input = e.target.name;

        if(input === "peittha") {
            this.setState({ peittha: value, kg: mmToMetric(value), lb: mmToImperial(value) });
        }

        if(input === "kg") {
            let peittha = metricToMm(value);
            this.setState({ peittha, kg: value, lb: mmToImperial(peittha) });
        }

        if(input === "lb") {
            let peittha = imperialToMm(value);
            this.setState({ peittha, kg: mmToMetric(peittha), lb: value });
        }
    }

    render() {
        return (
            <div>
                <Header />
                <input type="text"
                    value={this.state.peittha}
                    name="peittha"
                    onChange={this.handleChange} /> ပိဿာ <br />

                <input type="text"
                    value={this.state.kg}
                    name="kg"
                    onChange={this.handleChange}  /> ကီလို <br />

                <input type="text"
                    value={this.state.lb}
                    name="lb"
                    onChange={this.handleChange}  /> ပေါင် <br />
            </div>
        )
    }
}

export default App;
