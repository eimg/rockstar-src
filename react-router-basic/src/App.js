import React, { Component } from 'react';
import { BrowserRouter as Router, Link, Route, Redirect } from 'react-router-dom';

import Home from './Home';
import About from './About';
import Login from './Login';

import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';

import Drawer from '@material-ui/core/Drawer';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';

import BottomNavigation from '@material-ui/core/BottomNavigation';
import BottomNavigationAction from '@material-ui/core/BottomNavigationAction';
import RestoreIcon from '@material-ui/icons/Restore';
import FavoriteIcon from '@material-ui/icons/Favorite';
import LocationOnIcon from '@material-ui/icons/LocationOn';

const styles = {
    bannerButton: {
        position: 'absolute',
        top: 0,
        left: 0,
        background: 'rgba(0, 0, 0, 0.4)',
        color: 'white',
        padding: '4px 12px',
        borderRadius: '0 0 10px 0',
        border: '0 none'
    },
    banner: {
        width: 300,
        height: 200
    },
    hide: {
        display: 'none'
    }
}

class App extends Component {
    constructor() {
        super();
        this.state = {
            open: false,
            src: 'default.png',
            auth: false,
            content: 'none'
        }

        this.fileInput = React.createRef();
        this.form = React.createRef();
    }

    login = (username, password) => {
        fetch('http://localhost:8000/login', {
            method: 'post',
            headers: {
                "Content-Type": "application/json",
            },
            body: JSON.stringify({
                username, password
            })
        }).then(res => {
            if(res.ok)
                return res.json();

            alert("Auth fail");
        }).then(json => {
            this.setState({auth: json.token});

            fetch('http://localhost:8000/admin', {
                method: 'get',
                headers: {
                    'Authorization': 'Bearer ' + this.state.auth
                }
            }).then(res => res.json()).then(json => {
                this.setState({
                    content: json.content
                });
            });
        });
    }

    changeBanner = () => {
        const reader = new FileReader();
        reader.onload = (() => {
            let formData = new FormData( this.form.current );

            fetch('http://localhost:8000/upload', {
                method: 'post',
                body: formData
            });

            return (e) => {
                this.setState({
                    src: e.target.result
                });
            }
        })();

        reader.readAsDataURL(this.fileInput.current.files[0]);
    }

    render() {
        return (
            <Router>
                <div>
                    <AppBar position="static">
                        <Toolbar>
                            <IconButton color="inherit" onClick={() => {
                                this.setState({open: true})
                            }}>
                                <MenuIcon />
                            </IconButton>
                            <Typography variant="h6" color="inherit">
                                React Router
                            </Typography>
                        </Toolbar>
                    </AppBar>

                    <Drawer open={this.state.open} onClose={() => {
                        this.setState({open: false})
                    }}>
                        <div>
                            <form enctype="multipart/form-data" ref={this.form}>
                                <input type="file" name="banner" style={styles.hide} ref={this.fileInput} onChange={this.changeBanner} />
                            </form>

                            <button style={styles.bannerButton} onClick={() => {
                                this.fileInput.current.click();
                            }}>Change Banner</button>

                            <img src={this.state.src} alt="Banner" style={styles.banner} />
                        </div>

                        <List style={{width: 300}}>

                            <Link to="/">
                                <ListItem button onClick={() => {
                                    this.setState({open: false})
                                }}>
                                    <ListItemText primary="Home" />
                                </ListItem>
                            </Link>

                            <Link to="/about">
                                <ListItem button onClick={() => {
                                    this.setState({open: false})
                                }}>
                                    <ListItemText primary="About" />
                                </ListItem>
                            </Link>

                            <Link to="/login">
                                <ListItem button onClick={() => {
                                    this.setState({open: false})
                                }}>
                                    <ListItemText primary="Login" />
                                </ListItem>
                            </Link>
                        </List>
                    </Drawer>

                    <Route path="/" exact render={() => {
                        if(this.state.auth) {
                            return (<Home content={this.state.content} />)
                        } else {
                            return (<Redirect to="/login" />)
                        }
                    }} />

                    <Route path="/about" component={About} />

                    <Route path="/login" render={() => {
                        if(this.state.auth) {
                            return (<Redirect to="/" />)
                        } else {
                            return <Login login={this.login} />
                        }
                    }} />

                    <Route path="/more/:label" component={More} />

                    <BottomNavigation style={{
                        position: 'fixed',
                        bottom: 0,
                        width: '100%'
                    }}>
                        <Link to="/more/recent">
                            <BottomNavigationAction icon={<RestoreIcon />} />
                        </Link>
                        <Link to="/more/favorites">
                            <BottomNavigationAction icon={<FavoriteIcon />} />
                        </Link>
                        <Link to="/more/nearby">
                            <BottomNavigationAction icon={<LocationOnIcon />} />
                        </Link>
                    </BottomNavigation>
                </div>
            </Router>
        );
    }
}

const More = props => <div>{props.match.params.label}</div>

export default App;
