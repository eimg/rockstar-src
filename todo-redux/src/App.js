import React from 'react';
import Divider from '@material-ui/core/Divider';
import Snackbar from '@material-ui/core/Snackbar';

import Header from './Header';
import Add from './Add';
import Todo from './Todo';
import Done from './Done';

import { connect } from 'react-redux';

const api = 'http://206.189.42.134';

class App extends React.Component {
    render() {
        return (
            <div>

                <Header
                    clear={this.props.clear}
                    count={this.props.todo.filter((item) => {
                        return item.status === 0
                    }).length} />

                <Add handler={this.props.add} />

                <Todo
                    done={this.props.done}
                    remove={this.props.remove}
                    data={this.props.todo.filter((item) => item.status === 0)} />

                <Divider />

                <Done
                    undo={this.props.undo}
                    remove={this.props.remove}
                    data={this.props.todo.filter((item) => item.status === 1)} />

                <Snackbar
                    message="Server error"
                    open={this.props.apiError} />
            </div>
        )
    }
}

var ReduxApp = connect((state) => {
    return {
        todo: state.todo,
        apiError: state.apiError || false
    }
}, (dispatch) => {
    return {
        add: (subject) => {
            fetch(`${api}/tasks`, {
                method: 'post',
                headers: {
                    "Content-Type": "application/json",
                },
                body: JSON.stringify({
                    subject, status: 0
                })
            }).then((res) => res.json()).then((json) => {
                dispatch({ type: 'ADD', todo: json });
            });
        },
        remove: (_id) => {
            fetch(`${api}/tasks/${_id}`, { method: 'delete' });
            dispatch({ type: 'REMOVE', _id });
        },
        done: (_id) => {
            fetch(`${api}/tasks/${_id}`, {
                method: 'put',
                headers: {
                    "Content-Type": "application/json",
                },
                body: JSON.stringify({
                    status: 1
                })
            });

            dispatch({ type: 'DONE', _id });
        },
        undo: (_id) => {
            fetch(`${api}/tasks/${_id}`, {
                method: 'put',
                headers: {
                    "Content-Type": "application/json",
                },
                body: JSON.stringify({
                    status: 0
                })
            });

            dispatch({ type: 'UNDO', _id });
        },
        clear: () => {
            fetch(`${api}/tasks`, { method: 'delete' });
            dispatch({ type: 'CLEAR' });
        }
    }
})(App);

export default ReduxApp;
