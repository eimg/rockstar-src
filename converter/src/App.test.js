import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';

test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
});

test('<App /> renders correctly', () => {
    const app = renderer.create(<App />).toJSON();
    expect(app).toMatchSnapshot();
});
