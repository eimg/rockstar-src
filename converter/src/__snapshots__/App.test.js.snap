// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<div>
  <h1
    title="Unit Converter"
  >
    Converter
  </h1>
  <input
    name="peittha"
    onChange={[Function]}
    type="text"
    value={1}
  />
   ပိဿာ 
  <br />
  <input
    name="kg"
    onChange={[Function]}
    type="text"
    value={1.63}
  />
   ကီလို 
  <br />
  <input
    name="lb"
    onChange={[Function]}
    type="text"
    value={3.6}
  />
   ပေါင် 
  <br />
</div>
`;
